app/controllers/api/profile/password_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Api::Profile::PasswordController < ApiController
+  self.permitted_attrs = [:old_password, :new_password1, :new_password2]
+
+  before_action :authorize_action
+
+  def update
+    if password_params_valid?
+      current_user.update_password(old_password,
+                                   new_password)
+      add_info('flashes.session.new_password_set')
+      ok_status = 200
+    end
+    render_json(nil, ok_status || 400)
+  end
+
+  def password_params_valid?
+    return if current_user.is_a?(User::Api)
```

Comment:
maybe integrate this check into policy? so this would return a 403 in case an api user tries to access this endpoint ...
add a test case for this